### PR TITLE
feat: (Core) implement wizard summary

### LINKED ISSUE #3588: Wizard needs summary page documentation
We need an example that shows a summary page which can link back to prior wizard steps. **See the comment below for branch and explanation of what's blocked**

### PR DESCRIPTION
**Pull request by mikerodonnell89**
#### Please provide a link to the associated issue.

fixes #3588 

#### Please provide a brief summary of this pull request.

Adds the wizard summary page and full-screen wizard examples.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:280]: You can use `.style.removeProperty("display")` there

File: libs/core/src/lib/wizard/wizard-progress-bar/wizard-progress-bar.directive.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard-progress-bar/wizard-progress-bar.directive.ts:36]: You can use `.style.removeProperty("display")` there instead of storing `this.initDisplay`

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:273]: Coud you create variable from this? ex. `lastNonSummaryStep`, It's quite hard to read this condition

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:357]: Instead of this `if`, you can use filter at the beginning of array

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:272]: `this.steps.toArray()[this.steps.length - 2]` is used 6 times in code, maybe you can create some getter/method for pre latest step?

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:428]: I think you can feel free to use find() there instead o filter, It will return first found item. Also remove unnecesary `{}`

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:438]: Can you use `some()` there?

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
Hi @mikerodonnell89 Component works great, I just added some comments regarding the code

**Comment by JKMarkowski** [comment]
Also please add  `WizardModule` to `FundamentalNgxCoreModule`, because it's missing there

**Comment by mikerodonnell89** [comment]
> Also please add `WizardModule` into `FundamentalNgxCoreModule`, because it's missing there

I will do this in a separate PR so it can be merged ASAP

File: apps/docs/src/app/core/component-docs/wizard/wizard-docs.module.ts
Comment (mikerodonnell89) [reply apps/docs/src/app/core/component-docs/wizard/wizard-docs.module.ts:46]: Yes, the example uses the layout grid

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (mikerodonnell89) [reply libs/core/src/lib/wizard/wizard.component.ts:357]: This would be more code and harder to read in my opinion

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html:3]: Why don't you use the Dialog for full screen Wizard?

File: libs/core/src/lib/wizard/wizard-progress-bar/wizard-progress-bar.directive.ts
Comment (salarenko) [inline libs/core/src/lib/wizard/wizard-progress-bar/wizard-progress-bar.directive.ts:22]: Can we move this to host element and set based on `visible` property?

File: libs/core/src/lib/wizard/wizard-step/wizard-step.component.ts
Comment (salarenko) [inline libs/core/src/lib/wizard/wizard-step/wizard-step.component.ts:135]: Could you move these setup steps into separate methods?

**Review by salarenko — CHANGES_REQUESTED** [review]

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html
Comment (mikerodonnell89) [reply apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.html:3]: Well it is not meant to be used in a dialog, so I didn't want to give the impression that it is, so I decided to go the same route that the flexible column layout docs are used.  The mobile wizard is used in a dialog and I'll use a dialog in that example

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [reply libs/core/src/lib/wizard/wizard.component.ts:351]: Both are equal imo, let's can keep `if` then

**Review by JKMarkowski — COMMENTED** [review]
My comments have been addressed, LGTM, i will approve, when Michal's comments are addressed

File: libs/core/src/lib/wizard/wizard.component.ts
Comment (JKMarkowski) [inline libs/core/src/lib/wizard/wizard.component.ts:416]: Could you remove brackets from this find?

**Comment by stefanoScalzo** [comment]
The Edit button should be at the end of the page and aligned with the exit button

**Comment by mikerodonnell89** [comment]
> The Edit button should be at the end of the page and aligned with the exit button

Where are you seeing that?

**Review by JKMarkowski — APPROVED** [review]
LGTM

**Comment by mikerodonnell89** [comment]
> The Edit button should be at the end of the page and aligned with the exit button

Looks like this varies depending on which documentation you're looking at.  I have contacted the designers but have yet to receive a response

**Pull request merged**
